src/App.tsx

```diff
@@ -15,6 +24,37 @@ function App() {
     setSelectedGenre(genre);
   }
 
+  const handleSortChange = (newSort: string) => {
+    console.log("New sortOrder:", newSort);
+    setSelectedSort(newSort);
+  };
+
+  const handleMovieClick = (movie: Movie) => {
+    console.log("Clicked movie:", movie);
+    // Perform any action when a movie is clicked, such as showing details
+  };
+
+  const handleEditMovie = (movie: Movie) => {
+    console.log("Edit movie:", movie);
+    // Perform edit operation on the movie
+  };
+
+  const handleDeleteMovie = (movie: Movie) => {
+    console.log("Delete movie:", movie);
+    // Perform delete operation on the movie
+  };
+
+  // Sample movie data
+  const movie = {
```

Comment:
Here is the example of the data from the API:
Pay attention to the fields naming
  {
      "id": 337167,
      "title": "Fifty Shades Freed",
      "tagline": "Don't miss the climax",
      "vote_average": 6.1,
      "vote_count": 1195,
      "release_date": "2018-02-07",
      "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
      "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
      "budget": 55000000,
      "revenue": 136906000,
      "genres": [
          "Drama",
          "Romance"
      ],
      "runtime": 106
  },